src/main/java/com/direwolf20/mininggadgets/common/items/MiningGadget.java

```diff
@@ -472,16 +482,19 @@ else if (g > 0 && b == 1)
                 pos = lookingAt.getBlockPos().relative(side).relative(right);
 
             if (world.getMaxLocalRawBrightness(pos) <= 7 && world.getBlockState(pos).getMaterial() == Material.AIR) {
-                int energy = stack.getCapability(CapabilityEnergy.ENERGY).map(IEnergyStorage::getEnergyStored).orElse(0);
+                int energy = stack.getCapability(ForgeCapabilities.ENERGY).map(IEnergyStorage::getEnergyStored).orElse(0);
                 if (energy > Config.UPGRADECOST_LIGHT.get()) {
                     world.setBlockAndUpdate(pos, ModBlocks.MINERS_LIGHT.get().defaultBlockState());
-                    stack.getCapability(CapabilityEnergy.ENERGY).ifPresent(e -> e.receiveEnergy((Config.UPGRADECOST_LIGHT.get() * -1), false));
+                    stack.getCapability(ForgeCapabilities.ENERGY).ifPresent(e -> e.receiveEnergy((Config.UPGRADECOST_LIGHT.get() * -1), false));
                 }
             }
         }
     }
 
     public static int getEnergyCost(ItemStack stack) {
+        if (UpgradeTools.containsActiveUpgradeFromList(UpgradeTools.getUpgrades(stack), Upgrade.BATTERY_CREATIVE))
+            return 0;
+
         int cost = Config.MININGGADGET_BASECOST.get();
         List<Upgrade> upgrades = UpgradeTools.getActiveUpgrades(stack);
```

Comment:
I'd recommend checking the gagdet's upgrade here instead of the `containsActiveUpgradeFromList` above.